240 test accuracy of speaker and face identification on dutch debate data set

n400peanuts opened the pull request.

Commit:
scripts for creating different types of dataset

Commit:
subset algorithm updated to create the minimal duration possible;
;

n400peanuts committed:
added files for video evaluation

Commit:
folder to render the html file

Commit:
diarization performance

Commit:
create datataset script

n400peanuts committed:
Delete .DS_Store

Commit:
integrate feedback on video performance

Commit:
updates on video performance

Commit:
ciaoMerge branch '240-test-accuracy-of-speaker-and-face-identification-on-dutch-debate-data-set' of https://github.com/mexca/mexca-eval into 240-test-accuracy-of-speaker-and-face-identification-on-dutch-debate-data-set

n400peanuts commented:
@maltelueken evaluation of mexca's diarization pipeline has been pushed in the audio/debate/ folder in the quarto_mexca_performance_diarization.qmd notebook. Note also that I pushed also the evaluation of the video pipeline again (video/debate/) integrating your feedback into the quarto_mexca_performance_video.qmd notebook

I would recommend review this PR before our meeting on tuesday and leaving here all the comments.

maltelueken commented on audio/quarto_mexca_performance_diarization.qmd:
````diff
@@ -0,0 +1,378 @@
+embed_audio("output_wav/unbalanced/list_audio_2_60_unbalanced.wav")
+```
+
+To better investigate whether the **duration** of the **speaking** **time** of each **speaker** would affect performance, speakers' speaking duration is manipulated to follow an exponential distribution.
````
Maybe here it would be good to add a line here on why you chose the exponential distribution. It took be a bit to understand that this is to create an unbalanced data set.

maltelueken commented on audio/quarto_mexca_performance_diarization.qmd:
```diff
@@ -0,0 +1,378 @@
+  chunk_output_type: inline
+---
+
+Mexca's video subcomponent is based on [Pyannote.audio 2.0](https://github.com/pyannote/pyannote-audio). An open-source toolkit written in Python for speaker diarization. Pyannote.audio is based on PyTorch machine learning framework, and it comes with [pre-trained models](https://huggingface.co/pyannote/speaker-diarization) for speaker diarization, which we have used here.
```
*Audio* sub component.

maltelueken commented on audio/quarto_mexca_performance_diarization.qmd:
````diff
@@ -0,0 +1,378 @@
+
+```
+
+It is evident that the poor performance in those datapoints can be easily attributed to two speakers (Klaver and Wilders) and for 1 instance in Burger. Apart from Burger (that it might be just bad luck), this plot suggests that f1_score seems to be influenced more by the singular features of the speakers rather than the total number of speakers.
````
> that it might be just bad luck

I would choose a different phrasing here.

maltelueken commented on audio/audio_utils.R:
```diff
@@ -0,0 +1,231 @@
+    wav_file$n_speaker <- number_of_speakers
+    wav_file$balanced <- balanced
+    
+    write.csv(wav_file, file.path(outfolder, paste0("speaker_", number_of_speakers, "_duration_", max_duration, "_",ifelse(balanced == T, "balanced","unbalanced"), ".csv")))
```
```suggestion
    write.csv(wav_file, file.path(outfolder, paste0("speaker_", number_of_speakers, "_duration_", max_duration, "_",ifelse(balanced, "balanced","unbalanced"), ".csv")))
```

For logical variables, you don't need to test if they are equal to `TRUE`. This occurs a few times in this file.

Another suggestion for good practice: Instead of `T` and `F`, write out `TRUE` and `FALSE`. This is a bit more tedious to type but prevents (hard to find) bugs when you assign values to `T` and `F`, e.g., if you do `T = 0` beforehand.

n400peanuts replied:
what do you mean 'For logical variables, you don't need to test if they are equal to TRUE'? You mean that I can use binary numbers? e.g., TRUE/1 -- FALSE/0? 

for the second suggestion, I agree that this is less prone to bugs -- I'm adding it

maltelueken replied:
I mean that instead of writing `ifelse(balanced == TRUE, ..., ...)`, you can write `ifelse(balanced, ..., ...)`. You don't need to explicitly test whether logical variables are equal to `TRUE`, this will be alway done implicitly.

n400peanuts replied:
ahhhhh! This is great, yes I'll do that then. Thanks for the tip!

maltelueken commented on audio/quarto_mexca_performance_diarization.qmd:
```diff
@@ -0,0 +1,378 @@
+
+`Answer`: Yes. Performance decreases for speakers whose speaking time is over \~8 min.
+
+`Conclusion:` Our results suggest that the audio pipeline can be safely used for any number of speaker (up to the tested number of 7) without decreasing in performance, however seems likely to be affected by how long one speaks compared to the others. To be sure though, one needs to analyse also a more balanced dataset (e.g., whereby each speaker speaks the same amount of time).
```
I would not use the word *safely* here because the classifier completely misidentifies some speakers in (admittedly) rare cases.

n400peanuts replied:
I agree with your wording. I am going to change that also in the video notebook

maltelueken commented on audio/quarto_mexca_performance_diarization.qmd:
````diff
@@ -0,0 +1,378 @@
+head(F1_total)
+```
+
+## F1 score distribution
````
Maybe I missed this: But is the following analysis about the balanced or unbalanced data set (or both) and is there a difference between the two data sets?

n400peanuts replied:
Thanks for noticing this! this analysis is on the unbalanced dataset, as I thought this would be more informative. I have specified that now in the quarto notebook

maltelueken reviewed (commented):
The analysis looks good in general. I left a few comments that might need some clarification. I still need to look a bit more into the code and reproduce the analysis.

maltelueken commented:
I could reproduce the audio analysis well, nice. For the video, I ran into one issue which I comment in the file.

maltelueken commented on video/debate/quarto_mexca_performance_video.qmd:
````diff
@@ -0,0 +1,343 @@
+We add to mexca's output the column `shot_id` and `filename` indicating the shots that made each video in the file/`video_name` column. Moreover, we also add in the column `n` the order in which the shots were concatenated. This info is useful in order to understand who was displayed when.
+
+```{r}
+input_files <- c(list.files('total', full.names = T), list.files('contemporary', full.names = T))
````
In this chunk, it does not seem to find the `input_files`. Maybe this is just a path issue?

n400peanuts replied:
Nope! my mistake, It was supposed to load a folder that was not present. I loaded it now

maltelueken replied:
It works now, great!

Commit:
colab notebook to run mexca audio test

Commit:
mexca notebook audio test

Commit:
added total folder

Commit:
quarto notebook now specifies that analyses are about the unbalanced dataset

Commit:
removed `safely` within the conclusions in the audio quarto document

Commit:
removed `safely` within the conclusions of the video quarto notebook

Commit:
removed redundant logical statements

Commit:
added data folder

Commit:
polished files

Commit:
deleted contemporary notebook

Commit:
list of shots with their absolute path

Commit:
remove unused files

Commit:
polished python notebook to run mexca

Commit:
added README

Commit:
fixed wording in README

maltelueken reviewed (approved):
This looks great now, can be merged! Also great that you added a README!

EDIT: Apologies, I didn't see that you were still working on this!

Commit:
polished notebooks added figures

Commit:
ref audio rttm files

Commit:
polished folder

n400peanuts committed:
Delete .DS_Store

Commit:
redundant folder deleted

Commit:
solve conflicts
Merge branch '240-test-accuracy-of-speaker-and-face-identification-on-dutch-debate-data-set' of https://github.com/mexca/mexca-eval into 240-test-accuracy-of-speaker-and-face-identification-on-dutch-debate-data-set
